### ISSUE TITLE
Virus - False Positive

### ISSUE THREAD

**Issue by mzomparelli** [mzomparelli/zVirtualDesktop#266]
Well, it happened to me. I performed some testing and I think I have identified the false positive as the result of zVD updater exe being embedded inside zVD's exe and extracted on demand. This looks suspicious and I get it from the virus protection standpoint. The reason I do this is so that zVD can be contained within a single executable instead of being installed. I prefer it the way it is, but I may be forced to turn this into an installed program.

If you have any questions or concerns please do not hesitate to comment. Also, let me know if you would prefer zVD to be installed using an installer.

**Comment by MikeCress** [comment]
Michael, This issue is making using zvirtualdesktop almost impossible. Its a great program and I like portable exes as a principle but Norton and WIndows Defender are both stopping it now. An Installed version is probably needed...

**Comment by mzomparelli** [comment]
I'll work on getting to the bottom of this.

**Comment by mzomparelli** [comment]
After messing around with the settings for the obfuscation tool I'm using I think I can say it is the cause. While I investigate other tools for this purpose I have worked out some settings that pass VirusTotal. Please test a new build and let me know if your anti-virus still complains.

**Comment by MikeCress** [comment]
The latest insider build triggers a different objection from Norton.
Build for 20H2:
![Screenshot 2020-12-06 080728](https://user-images.githubusercontent.com/75562335/101316637-c943bc80-3854-11eb-8303-76182f2ea587.jpg)
Current Insider build:
![Screenshot 2020-12-07 062039](https://user-images.githubusercontent.com/75562335/101316643-ccd74380-3854-11eb-99a6-dd5567606264.jpg)
Could be progress...
I'll try temporarily turning off Norton and see if Defender will allow the latest build to run...

**Comment by MikeCress** [comment]
Using current insider build:
No surprise that turning off Norton allows the unzipping to happen.
Windows Defender still doesnt like the updater but at least you can tell it to run it anyway.
But this build doesnt seem to run on my Windows 20H2 after all that, anyway...

**Comment by MikeCress** [comment]
Further news:
The current build for 20H2 gets past Norton (it only triggers a warning that I can ignore)
Running the exe then makes Defender warn against running it.
Ignoring Defender and running it anyway, you have to give it admin level access (no problem) and then nothing happens and theres nothing identifiable as zvirtualdesktop to be found in task manager...
I think thats all the help I'm going to be able to give for this iteration Michael (if it is help!)

**Comment by mzomparelli** [comment]
Yeah something going on with this obfuscation tool I'm using. Some builds run and some don't. I just published all new builds. Hopefully this goes away with a different obfuscation tool.

Thanks for the great info though. It's really helpful.

**Comment by mzomparelli** [comment]
The WS.Reputation.1 appears to be just that. A not well known app. The other one concerns me though.

https://community.norton.com/en/forums/clarification-wsreputation1-detection

**Comment by MikeCress** [comment]
I've tried the 20H2 build.  It extracts without any trouble from Norton, so that's progress... Running it causes Windows Defender to warn but you can ignore that, then Norton complains that its a new unknown program so you can ignore that, then it puts an icon in the taskbar tray, so it looks good. But that icon vanishes as soon as my mouse hovers over it and there's nothing identifiably zVD running according to task manager.  So that's closer than we were before but still no cigar!

**Comment by MikeCress** [comment]
Just to add, I've run it a few times and the icon it puts in the taskbar tray is sometimes an accurate identifier of the desktop I'm on, sometimes just the zVD program icon. Just in case this helps...

**Comment by mzomparelli** [comment]
Try running again. Sometimes there is crash on startup of the app. I think it might be the mouse hook causing that and I'm trying to resolve it. To be safe, try not moving the mouse while zVD is loading and the crash should be avoided.

**Comment by MikeCress** [comment]
I've tried that, and done a restart and tried it again, but its still the same.
Im pretty sure Ive got a clean up-to-date 20H2 Win10 system but I have had a glitch or two (drivers I imagine) since the last MS update so maybe its my system...

**Comment by mzomparelli** [comment]
I did discover another issue in the tool I use for obfuscation last night. The bug had caused me to publish bad builds. If you haven't downloaded https://zVirtualDesktop.com since late last night then please try that getting a new download and let me know.

**Comment by MikeCress** [comment]
Michael, I've done that with the same effect.  But I've also taken the zVD exe which works fine on another machine with windows 20H2 and moved it to my main machine.  On my main machine it has exactly the same problem of starting and immediately crashing.  So I think it must be some conflict in my system and I apologise for being a red herring in at least some of this.  Maybe I will see if I can get to the bottom of the post-windows update mouse glitch on my system and then try zVD on it again.  Oh how I wish I had never agreed to let the zVD updater do its thing - zVD was working fine with 20H2 until I did that!  Anyway, thank you and don't give up on zVD.  Its a unique bit of software these days and I miss it!

**Comment by mzomparelli** [comment]
It should be working fine now. I'm running 20H2 on my dev machine. I'm wondering if maybe with my recent complications with the obfuscator tool that maybe one of my more recent previous builds has corrupted your settings fine. Can you try deleting the settings file and then running zVD again. Maybe backup the settings file first in case it is not the issue.

%appdata%\zVirtualDesktop\zVirtualDesktop_Settings.bin

**Comment by MikeCress** [comment]
I wish, Michael... I think I've fixed my mouse driver problem (time will tell) and I've run sfc to make sure my system is as good as it can be short of doing something drastic...  So got back to zVD...
Deleting the settings file meant that, upon startup, I got the updater asking if I wanted to update (a surprise because I was running the build I know works on my other Win10 20H2 machine and doesn't ask to update there!) but now with some terms and conditions etc. behind it.
I just watched this and after a few seconds it all disappeared and, just as before, the desktop icon appeared in my system tray but vanished as soon as my mouse rolled over it.  Watching task manager, showed the software starting and then stopping when the updater disappeared...
Tbh, while I am happy to try anything else you come up with, if you're interested, I think it must be something to do with my machine and I am at the point of deciding just to live without zVD on this machine.  Annoying, but hey...

**Comment by mzomparelli** [comment]
please run "Winver" to confirm what build you are on. Paste your window here so I can view and make sure you get the correct link to download.

![image](https://user-images.githubusercontent.com/10564234/101990189-1b2c7e00-3c6b-11eb-9f26-49364e7cf3b1.png)

**Comment by MikeCress** [comment]
Thanks for sticking with it, Michael.

![Screenshot 2020-12-12 183504](https://user-images.githubusercontent.com/75562335/101992137-e93a0c80-3ca8-11eb-96fa-abcfe6cb2463.jpg)

**Comment by mzomparelli** [comment]
Deleting the settings file should not have prompted to update. Try deleting the file again and downloading using this link.

https://zomp.co/files.aspx?id=zvd

**Comment by mzomparelli** [comment]
What might cause the update to be triggered could be if zVD is unable to determine or is incorrectly determining your Windows 10 version. I'm not sure what would cause this, but if your still having issues after my previous post then let's try a custom app that tries to determine your OS version so that we can rule that out.

**Comment by MikeCress** [comment]
Running the version in the link you sent did not trigger an update.  I had to tell Defender to allow it to run, and then Norton, and then tell Norton again when it said the digital signature on the file was invalid. Then - exactly as before, zVD started, put its icon in my system tray and crashed so the icon disappeared when I pointed at it in hope...

**Comment by mzomparelli** [comment]
Can you pull the stack trace from event viewer when it crashes? Also, something else to try is deleting the settings file so that a new one can be created at launch. It could be an issue with the settings file.
